tools/analyzers/writefile/analyzer.go

```diff
@@ -0,0 +1,101 @@
+// Package writefile implements a static analyzer to ensure that Prysm does not
+// use ioutil.MkdirAll or os.WriteFile as they are unsafe when it comes to guaranteeing
+// file permissions and not overriding existing permissions.
+package writefile
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+// Doc explaining the tool.
+const Doc = "Tool to enforce usage of Prysm's internal file-writing utils instead of os.MkdirAll or ioutil.WriteFile"
+
+var errUnsafePackage = errors.New(
+	"os and ioutil dir and file writing functions are not permissions-safe, use shared/fileutil",
+)
+
+// Analyzer runs static analysis.
+var Analyzer = &analysis.Analyzer{
+	Name:     "writefile",
+	Doc:      Doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	inspect, ok := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	if !ok {
+		return nil, errors.New("analyzer is not type *inspector.Inspector")
+	}
+
+	nodeFilter := []ast.Node{
+		(*ast.File)(nil),
+		(*ast.ImportSpec)(nil),
+		(*ast.CallExpr)(nil),
+	}
+
+	aliases := make(map[string]string)
+	disallowedFns := []string{"MkdirAll", "WriteFile"}
+
+	inspect.Preorder(nodeFilter, func(node ast.Node) {
+		switch stmt := node.(type) {
+		case *ast.File:
+			// Reset aliases (per file).
+			aliases = make(map[string]string)
+		case *ast.ImportSpec:
+			// Collect aliases.
+			pkg := stmt.Path.Value
+			if pkg == "\"os\"" {
+				if stmt.Name != nil {
+					aliases[stmt.Name.Name] = stmt.Path.Value
```

Comment:
You can replace `stmt.Path.Value` with `pkg` everywhere inside both conditionals.